Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -517,7 +517,7 @@ public struct FileRuleDescription {
         .init(
             rule: .header,
             toolsVersion: .minimumRequired,
-            fileTypes: ["h"]
```

Comment:
Xcode considers even more extensions as headers: hp, hpp, hxx, h++, H, pch++, hh, ipp, h, pch

Not sure if we need the PCH ones, but we should probably add the other ones, especially `hxx` is rather common.

Reply:
The new list looks good, except that I'm not sure about `def`.  Maybe that's a common one and just unknown to me.  I don't think Xcode treats that one as a header.

Reply:
`.def` files are used in the Swift toolchain itself and are included as any other header there: https://github.com/apple/swift/blob/master/include/swift/ABI/MetadataKind.def

I've added it as I had problems compiling a few Swift packages that reuse some of those headers. Currently SwiftPM recognizes those files as resources and requires those to be added to the list of resources explicitly in 5.3 package manifests.

Reply:
I see — thanks for the context.  Since any file with any suffix can be included in a C header, I support that we have a bit of an open-ended problem here.  I guess conceptually, any file that's included from a header should be exempted from being treated as a resource, but I don't see a good way to implement that robustly (certainly not using the file system scanning approach that SwiftPM uses today).

Reply:
Thank you for looking into this, would you like me to remove `.def` from the list after all?

Reply:
I'm kind of on the fence about it — I totally see the motivation, but on the other hand, it's an open-ended set.  I think maybe we could exclude `.def` and keep the party with Xcode at the moment, and then have a broader discussion about how to handle a more open-ended set.

Would you mind adding or extending a unit test, though?  That would be great.  Thanks also for doing this!